x/bsonx/bsoncore/value.go

```diff
@@ -57,10 +57,10 @@ func (v Value) IsNumber() bool {
 	}
 }
 
+// TODO(skriptble): Add support for Decimal128.
```

Comment:
Optional: Consider completely removing these TODOs instead of moving them.

I'm not sure we should support decimal128-to-integer conversion. The main motivation for convenient conversions between BSON float/int types is to interoperate better with dynamically-typed languages that may write different numeric types for the same value (e.g. Javascript or Python). However, AFAIK no officially supported dynamically-typed languages have a native decimal128 implementation, so that silent mixing of types would never include decimal128. Basically, to end up with a decimal128 value in MongoDB, users have to explicitly use that type. Unless someone specifically requests it, there's no obvious use case for supporting decimal128-to-integer conversion.